Sources/OpenCombine/Published.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  Published.swift
+//  OpenCombine
+//
+//  Created by Евгений Богомолов on 01/09/2019.
+//
+
+#if swift(>=5.1)
+@propertyWrapper public struct Published<Value> {
+
+    /// Initialize the storage of the Published
+    /// property as well as the corresponding `Publisher`.
+    public init(initialValue: Value) {
+        self.projectedValue = .init(initialValue)
+    }
+
+    public struct Publisher: OpenCombine.Publisher {
+
+        /// The kind of values published by this publisher.
+        public typealias Output = Value
+
+        /// The kind of errors this publisher might publish.
+        ///
+        /// Use `Never` if this `Publisher` does not publish errors.
+        public typealias Failure = Never
+
+        /// This function is called to attach the specified
+        /// `Subscriber` to this `Publisher` by `subscribe(_:)`
+        ///
+        /// - SeeAlso: `subscribe(_:)`
+        /// - Parameters:
+        ///     - subscriber: The subscriber to attach to this `Publisher`.
+        ///                   once attached it can begin to receive values.
+        public func receive<SubscriberType>
+            (subscriber: SubscriberType)
+            where Value == SubscriberType.Input,
+            SubscriberType: Subscriber,
+            SubscriberType.Failure == Published<Value>.Publisher.Failure
+        {
+            subject.subscribe(subscriber)
+        }
+
+        fileprivate let subject: OpenCombine.CurrentValueSubject<Value, Never>
+
+        fileprivate init(_ output: Output) {
+            self.subject = .init(output)
+        }
+
+        fileprivate func send(_ input: Output) {
+            subject.send(input)
+        }
+    }
+
+    /// The property that can be accessed with the
+    /// `$` syntax and allows access to the `Publisher`
+    public let projectedValue: Published<Value>.Publisher
```

Comment:
This should be lazy var. I've checked, in Combine it **is** lazy var. This publisher should not be created until somebody reads `projectedValue` for the first time. So we need to make it lazy, like it was before:

```swift
private var _value: Value
public private(set) lazy var projectedValue = Publisher(self._value)
```

You can use the `dump` function from the standard library to explore the behavior of Combine:

```swift
import Combine

var published = Published(initialValue: 42)
dump(published)
// Prints:
// ▿ Combine.Published<Swift.Int>
//   - value: 42
//   - publisher: nil
//   - objectWillChange: nil

_ = published.projectedValue

dump(published)
// Prints:
// ▿ Combine.Published<Swift.Int>
//   - value: 42
//   ▹ publisher: Optional(Combine.Published<Swift.Int>.Publisher(subject: Combine.CurrentValueSubject<Swift.Int, Swift.Never>))
  - objectWillChange: nil
```

Reply:
I've checked this too, and it's turned out that projectedValue not used lazy storage, so I created publisher property which projectedValue wraps.

Reply:
Nice!